### PR TITLE
Allow redirect matching to include/exclude query string parameters when matching requests

### PR DESCRIPTION
**Pull request by alunwcom**
Previous behaviour always added the query string to the request path before regex matching was performed. This behaviour has been revised so that by default the query string is not appended to the path - with the option to include the query string by using the new 'includeParams: {true|false}' configuration option.

In order to facilitate testing, the worker script has been converted to ES6 module format to facilitate unit testing with Miniflare and Jest. Worker unit tests have also been added.

@BigBlueHat Please review whether this is suitable for PR, or whether modification of the changes are required.

### CODE REVIEW

File: tests/lib/shared.test.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ describe('has Conflicting DNS Records', () => {
 describe('has DNS Record', () => {
   it('should return true if the record exists in the current DNS', () => {
     const result = hasDNSRecord(required_dns_records, dns_line);
-    console.log(result);
+    // console.log(result);
```
Comment (BigBlueHat) [inline tests/lib/shared.test.js:103]: Saving this for later? 😉

File: tests/worker/bar.com.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+name: bar.com
+redirects: 
+
+# currently yaml files require at least one rule otherwise script breaks
+- from: /xyz.html
+  to: https://foo.com/xyz.html
+
```
Comment (BigBlueHat) [inline tests/worker/bar.com.yaml:7]: 👻 spaces need exorcism. 😉

**Review by BigBlueHat — CHANGES_REQUESTED** [review]
Super tiny tweaks. I'm also testing this locally, atm.

**Comment by BigBlueHat** [comment]
@alunwcom I'm getting `SyntaxError: Cannot use import statement outside a module` which may be why this is still a Draft?

**Comment by BigBlueHat** [comment]
> > @alunwcom I'm getting `SyntaxError: Cannot use import statement outside a module` which may be why this is still a Draft?
> 
> @BigBlueHat Sorry - I need to update the README here, as to run the tests you need to declare an environment variable for ES6 modules...
> 
> ```
> > $env:NODE_OPTIONS="--experimental-vm-modules npx jest"
> ```
> 
> or equivalent statement on *nix

If it's required to run the tests, you could just put it into `npm run test`. There should be a cross-platform friendly way to do the options stuff...I think you just use the all caps key/value thing without `$env`.

**Comment by alunwcom** [comment]
@BigBlueHat have addressed the issues raised so far ;) ... anything else?

**Review by BigBlueHat — APPROVED** [review]

**Comment by alunwcom** [comment]
@BigBlueHat I think you'll have to merge this - I don't appear to have access to do so....

**Comment by BigBlueHat** [comment]
> @BigBlueHat I think you'll have to merge this - I don't appear to have access to do so....

You do now. 😁

**Pull request merged**